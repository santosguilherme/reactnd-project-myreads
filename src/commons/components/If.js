const If = ({test, children}) => (test ? children : false);

export default If;

// TODO - ver se vai precisar mesmo