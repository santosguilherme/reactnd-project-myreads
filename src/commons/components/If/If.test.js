import React from 'react';
import {shallow} from 'enzyme';
import If from './If';

it('renders without crashing', () => {
    const wrapper = shallow(<If/>);

    expect(wrapper).toMatchSnapshot();
});

it('testando o restante da condicional', () => {
    shallow(
        <If test={true}>
            <span>Test</span>
        </If>
    );
});