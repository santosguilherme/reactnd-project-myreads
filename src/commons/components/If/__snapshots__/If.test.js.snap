// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <If />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
